sshd-sftp/src/main/java/org/apache/sshd/client/subsystem/sftp/impl/SftpRemotePathChannel.java

```diff
@@ -410,18 +408,23 @@ public long transferFrom(ReadableByteChannel src, long position, long count) thr
             try {
                 beginBlocking("transferFrom");
 
+                SftpOutputStreamAsync output = new SftpOutputStreamAsync(
+                        (AbstractSftpClient) sftp,
+                        copySize, getRemotePath(), handle);
```

Comment:
Shouldn't this be protected by a try-with-resource block (or similar) that closes it automatically when no longer needed (or exception thrown) ? I  see the comment that says not to close - but what if an exception is thrown ?

Reply:
Same as above.